### PR TITLE
AbstractKVS.internalTableName avoids expensive regex matcher

### PR DESCRIPTION
**Pull request by schlosna**
## General
**Before this PR**:
`com.palantir.atlasdb.keyvalue.impl.AbstractKeyValueService#internalTableName` used expensive regex to parse `TableReference` qualified name into internal name

image

**After this PR**:

==COMMIT_MSG==
AbstractKVS.internalTableName avoids expensive regex matcher
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**: Added tests that exercised existing implementation, then updated implementation.

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Review by mdaudali — COMMENTED** [review]
Thanks for the contribution! Just one requested change, for when the table name is empty but a namespace is present (as this technically was allowed until a very recent PR: #6146)

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java [hunk 1]
```diff
@@ -187,7 +184,11 @@ public static String internalTableName(TableReference tableRef) {
         if (tableName.startsWith("_")) {
             return tableName;
         }
-        return PERIOD_REGEX.matcher(tableName).replaceFirst("__");
+        int periodIndex = tableName.indexOf('.');
+        if (periodIndex > -1 && periodIndex != tableName.length() - 1) {
```
[the remark this answers is not in the document]
Comment (schlosna) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java]: Updated and added tests that cover empty table name, verified these pass before merging in the change banning empty table names.

**Comment by schlosna** [comment]
Thanks for the review @mdaudali . I updated the implementation to handle empty table names and improved tests to cover these cases.

**Review by mdaudali — APPROVED** [review]
lgtm, thank you!

**Pull request merged**